### PR TITLE
fix(dop,pipeline): Solve the problem that the data is incorrect after the execution of the cron pipeline

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What this PR does / why we need it:
Solve the problem that the state data is incorrect after the execution of the timing pipeline

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMTMxLDEwOTAsMTExNV0sImFzc2lnbmVlIjpbIjEwMDEyNjEiXX0%3D&id=289092&iterationID=1115&pId=0&type=BUG)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |      Solve the problem that the state data is incorrect after the execution of the timing pipeline        |
| 🇨🇳 中文    |      解决定时流水线执行后状态数据不正确问题        |

#### Need cherry-pick to release versions?
Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sfwn** [comment]
update latest running pipeline id inside `pipeline run` method in pipeline module.

File: modules/pipeline/services/pipelinesvc/run.go [hunk 1]
```diff
@@ -133,9 +135,44 @@ func (s *PipelineSvc) RunPipeline(req *apistructs.PipelineRunRequest) (*spec.Pip
 	// send to pipengine reconciler
 	s.engine.Send(p.ID)
 
+	// update pipeline definition
+	if err = s.updatePipelineDefinition(p); err != nil {
+		logrus.Errorf("failed to updatePipelineDefinition,err: %s", err.Error())
```
Comment (sfwn) [inline modules/pipeline/services/pipelinesvc/run.go:140]: `...o, err: ` pay attention on format please.
and you should print definition id and pipeline id together.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/services/pipelinesvc/run.go (hunk 1 above)
Comment (littlejiancc) [reply modules/pipeline/services/pipelinesvc/run.go]: done

File: cmd/actionagent/main_test.go [hunk 2]
```diff
@@ -1,65 +0,0 @@
-// Copyright (c) 2021 Terminus, Inc.
```
Comment (sfwn) [inline cmd/actionagent/main_test.go:1]: I think this file is belong to another pr.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sfwn — APPROVED** [review]
